bangazonapi/views/order.py

```diff
@@ -111,7 +111,7 @@ def update(self, request, pk=None):
         """
         customer = Customer.objects.get(user=request.auth.user)
         order = Order.objects.get(pk=pk, customer=customer)
-        order.payment_type = request.data["payment_type"]
+        order.payment_type_id = request.data["payment_type"]
```

Comment:
Do not directly set the foreign key on an object. Follow the company convention of validating the `payment_type_id` sent in the request by using the `.get()` method and catching the `DoesNotExist` method. If it is valid, set the `payment_type` property to the instance.

Reply:
updated with objects!